### PR TITLE
Implement vod storage experiment/feature toggle

### PR DESCRIPTION
**Pull request by mjh1**
**What does this pull request do? Explain your changes. (required)**

Allows us to switch over to a different storage provider in a gradual way.

**How did you test each of these updates (required)**

Tested on my local dev environment.

**Checklist**

- [x] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/controllers/asset.ts [hunk 1]
```diff
@@ -74,18 +77,32 @@ function isPrivatePlaybackPolicy(playbackPolicy: PlaybackPolicy) {
   return true;
 }
 
-function defaultObjectStoreId(
-  { config, body }: Request,
+const secondaryStorageExperiment = "secondary_vod_storage";
```
Comment (leszko) [inline packages/api/src/controllers/asset.ts]: Super nit: I think the experiments were designed to use in the URL, so maybe `secondaryVodStorage` or `secondary-vod-storage`?

File: packages/api/src/controllers/asset.ts [hunk 2]
```diff
@@ -74,18 +77,32 @@ function isPrivatePlaybackPolicy(playbackPolicy: PlaybackPolicy) {
...
+    user.id
+  );
+  const secondaryjObjectStoreId = secondaryStorageEnabled
+    ? config.secondaryVodObjectStoreId
+    : undefined;
+  return (
+    secondaryjObjectStoreId ||
+    config.vodCatalystObjectStoreId ||
+    config.vodObjectStoreId
+  );
```
Comment (leszko) [inline packages/api/src/controllers/asset.ts]:
```suggestion
const secondaryObjectStoreId = secondaryStorageEnabled && config.secondaryVodObjectStoreId;

return secondaryObjectStoreId || config.vodCatalystObjectStoreId || config.vodObjectStoreId;
```

Nit

File: packages/api/src/controllers/asset.ts [hunk 3]
```diff
@@ -74,18 +77,32 @@ function isPrivatePlaybackPolicy(playbackPolicy: PlaybackPolicy) {
...
   if (isPrivatePlaybackPolicy(body.playbackPolicy)) {
     return config.vodCatalystPrivateAssetsObjectStoreId;
   }
-  return config.vodCatalystObjectStoreId || config.vodObjectStoreId;
+
+  const secondaryStorageEnabled = await isExperimentSubject(
+    secondaryStorageExperiment,
+    user.id
+  );
+  const secondaryjObjectStoreId = secondaryStorageEnabled
```
Comment (leszko) [inline packages/api/src/controllers/asset.ts]: Typo `secondaryjObjectStoreId` => `secondaryObjectStoreId`

**Review by leszko — APPROVED** [review]
Added some minor comments. Other than that, LGTM 👍

File: packages/api/src/controllers/asset.ts (hunk 2 above)
Comment (mjh1) [reply packages/api/src/controllers/asset.ts]: `yarn run prettier` is multilining everything unfortunately but the `&&` is perfect thanks

**Review by thomshutt — APPROVED** [review]

**Pull request merged**